sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,92 +16,285 @@
+              "please check if ALLUXIO_HOME is set correctly")
+        } finally {
+          if (buffered_source != null) buffered_source.close
+        }
+
+        if (alluxio_master == null) {
+          throw new RuntimeException(
+            s"Can't find alluxio.master.hostname from $alluxioHome/conf/alluxio-site.properties.")
+        }
+        alluxioMasterHost = Some(alluxio_master + ":" + alluxio_port)
+        // load mounted point by call Alluxio mount command.
+        val (ret, output) = runAlluxioCmd("fs mount")
+        if (ret == 0) {
+          // parse the output, E.g.
+          // s3a://bucket-foo/        on  /bucket-foo
+          // s3a://bucket-another/    on  /bucket-another
+          // /local_path              on  /
+          for (line <- output) {
+            val items = line.trim.split(" +")
+            logDebug(line)
+            if (items.length >= 3) {
+              // if the first item contains the "://", it means it's a remote path.
+              // record it as a mounted point
+              if (items(0).contains("://")) {
+                mountedBuckets(items(2)) = items(0)
+                logInfo(s"Found mounted bucket ${items(0)} to ${items(2)}")
+              }
+            }
+          }
+        } else {
+          logWarning(s"Failed to run alluxio fs mount $ret")
+        }
+        isInit = true
+      }
+    }
+  }
+
+  // The path should be like s3://bucket/... or s3a://bucket/...
+  private def getSchemeAndBucketFromPath(path: String) : (String, String) = {
+    val i = path.split("://")
+    val scheme = i(0)
+    if (i.length <= 1) {
+      throw new RuntimeException(s"path $path is not expected for Alluxio auto mount")
+    }
+    val bucket = i(1).split("/")(0)
+    (scheme, bucket)
+  }
+
+  private def runAlluxioCmd(param : String) : (Int,
+    scala.collection.mutable.ArrayBuffer[String]) = {
+    val params = alluxioCmd.tails.collect {
+        case Seq(first, _, _*) => first
+        case Seq(last) => last + " " + param
+      }.toSeq
+    val out : scala.collection.mutable.ArrayBuffer[String] =
+      new scala.collection.mutable.ArrayBuffer[String](10)
+    val ret = if (params.length == 1) {
+      // if alluxioCmd is like "alluxio fs mount", run by Process(String)
+      Process(params.last).!(ProcessLogger(out += _, _ => Unit))
+    } else {
+      // if alluxioCmd has multiple strings in the seq like "su ubuntu -c 'alluxio fs mount'",
+      // need to run by Process(Seq[String])
+      Process(params).!(ProcessLogger(out += _, _ => Unit))
+    }
+    (ret, out)
+  }
+
+  // path is like "s3://foo/test...", it mounts bucket "foo" by calling the alluxio CLI
+  // And we'll append --option to set access_key and secret_key if existing.
+  // Suppose the key doesn't exist when using like Databricks's instance profile
+  private def autoMountBucket(scheme: String, bucket: String,
+                              access_key: Option[String],
```

Comment:
spacing is still off, parameters wrapping needs to be indented 4 spaces

Reply:
Fixed.